diff --git a/google_analytics_views/google_analytics_views.module b/google_analytics_views/google_analytics_views.module
index 0344ef2..60ba620 100644
--- a/google_analytics_views/google_analytics_views.module
+++ b/google_analytics_views/google_analytics_views.module
@@ -604,6 +604,209 @@ function google_analytics_views_get_fields() {
     'name' => 'ga:dayOfWeek',
     'description' => 'The day of the week. A one-digit number from 0 (Sunday) to 6 (Monday).',
   ),
+  'socialNetwork'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Traffic Sources',
+    'name' => 'ga:socialNetwork',
+    'description' => 'A string representing the social network name. This can be related to the referring social network for traffic sources, or to the social network for social data hub activities. E.g. Google+, Blogger, reddit, etc.',
+  ),
+  'hasSocialSourceReferral'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Traffic Sources',
+    'name' => 'ga:hasSocialSourceReferral',
+    'description' => ' Indicates visits that arrived to the site from a social source (i.e. A social network such as Google+, Facebook, Twitter, etc.). The possible values are Yes or No, where the first letter must be capitalized.',
+  ),
+  'socialActivityDisplayName'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Activities',
+    'name' => 'ga:socialActivityDisplayName',
+    'description' => 'For a social data hub activity, this string represents the title of the social activity posted by the social network user.',
+  ),
+  'socialActivityEndorsingUrl'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Activities',
+    'name' => 'ga:socialActivityEndorsingUrl',
+    'description' => 'For a social data hub activity, this string represents the URL of the social activity (e.g. the Google+ post URL, the blog comment URL, etc.)',
+  ),
+  'socialActivityPost'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Activities',
+    'name' => 'ga:socialActivityPost',
+    'description' => 'For a social data hub activity, this string represents the content of the social activity posted by the social network user (e.g. The message content of a Google+ post)',
+  ),
+  'socialActivityTimestamp'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Activities',
+    'name' => 'ga:socialActivityTimestamp',
+    'description' => 'For a social data hub activity, this string represents when the social activity occurred on the social network.',
+  ),
+  'socialActivityUserPhotoUrl'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Activities',
+    'name' => 'ga:socialActivityUserPhotoUrl',
+    'description' => ' For a social data hub activity, this string represents the URL of the photo associated with the user's social network profile.',
+  ),
+  'socialActivityUserProfileUrl'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Activities',
+    'name' => 'ga:socialActivityUserProfileUrl',
+    'description' => 'For a social data hub activity, this string represents the URL of the associated user's social network profile.',
+  ),
+  'socialActivityContentUrl'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Activities',
+    'name' => 'ga:socialActivityContentUrl',
+    'description' => 'For a social data hub activity, this string represents the URL shared by the associated social network user.',
+  ),
+  'socialActivityTagsSummary'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Activities',
+    'name' => 'ga:socialActivityTagsSummary',
+    'description' => 'For a social data hub activity, this is a comma-separated set of tags associated with the social activity.',
+  ),
+  'socialActivityUserHandle'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Activities',
+    'name' => 'ga:socialActivityUserHandle',
+    'description' => 'For a social data hub activity, this string represents the social network handle (i.e. name) of the user who initiated the social activity.',
+  ),
+  'socialActivityAction'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Activities',
+    'name' => 'ga:socialActivityAction',
+    'description' => 'For a social data hub activity, this string represents the type of social action associated with the activity (e.g. vote, comment, +1, etc.).',
+  ),
+  'socialActivityNetworkAction'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Activities',
+    'name' => 'ga:socialActivityNetworkAction',
+    'description' => 'For a social data hub activity, this string represents the type of social action and the social network where the activity originated.',
+  ),
+  'socialInteractionAction'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Interactions',
+    'name' => 'ga:socialInteractionAction',
+    'description' => 'For social interactions, a string representing the social action being tracked (e.g. +1, like, bookmark).',
+  ),
+  'socialInteractionNetwork'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Interactions',
+    'name' => 'ga:socialInteractionNetwork',
+    'description' => ' For social interactions, a string representing the social network being tracked (e.g. Google, Facebook, Twitter, LinkedIn).',
+  ),
+  'socialInteractionNetworkAction'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Interactions',
+    'name' => 'ga:socialInteractionNetworkAction',
+    'description' => 'For social interactions, a string representing the concatenation of the ga:socialInteractionNetwork and ga:socialInteractionAction being tracked (e.g. Google: +1, Facebook: like).',
+  ),
+  'socialInteractionTarget'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Social Interactions',
+    'name' => 'ga:socialInteractionTarget',
+    'description' => 'For social interactions, a string representing the URL (or resource) which receives the social network action.',
+  ),
+  'pagePathLevel1'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Page Tracking',
+    'name' => 'ga:pagePathLevel1',
+    'description' => 'This dimension rolls up all the pagePaths in the first hierarchical level in pagePath.',
+  ),
+  'pagePathLevel2'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Page Tracking',
+    'name' => 'ga:pagePathLevel2',
+    'description' => 'This dimension rolls up all the pagePaths in the second hierarchical level in pagePath.',
+  ),
+  'pagePathLevel3'=>
+  array(
+    'type' => 'dimension',
+    'group' => 'Page Tracking',
+    'name' => 'ga:pagePathLevel3',
+    'description' => 'This dimension rolls up all the pagePaths in the third hierarchical level in pagePath.',
+  ),
+  'pagePathLevel4' =>
+  array(
+    'type' => 'dimension',
+    'group' => 'Page Tracking',
+    'name' => 'ga:pagePathLevel4',
+    'description' => 'This dimension rolls up all the pagePaths in the fourth hierarchical level in pagePath. All the additional levels in the pagePath hierarchy are also rolled up in this dimension.',
+  ),
+  'userTimingCategory' =>
+  array(
+    'type' => 'dimension',
+    'group' => 'User Timings',
+    'name' => 'ga:userTimingCategory',
+    'description' => 'A string for categorizing all user timing variables into logical groups for easier reporting purposes).',
+  ),
+  'userTimingLabel' =>
+  array(
+    'type' => 'dimension',
+    'group' => 'User Timings',
+    'name' => 'ga:userTimingLabel',
+    'description' => 'A string to indicate the name of the action of the resource being tracked.',
+  ),
+  'userTimingVariable' =>
+  array(
+    'type' => 'dimension',
+    'group' => 'User Timings',
+    'name' => 'ga:userTimingVariable',
+    'description' => 'A string that can be used to add flexibility in visualizing user timings in the reports.',
+  ),
+  'userTimingValue' =>
+  array(
+    'type' => 'dimension',
+    'group' => 'User Timings',
+    'name' => 'ga:userTimingValue',
+    'description' => 'The number of milliseconds in elapsed time sent to Google Analytics.',
+  ),
+  'mobileDeviceBranding' =>
+  array(
+    'type' => 'dimension',
+    'group' => 'Device / Platform',
+    'name' => 'ga:mobileDeviceBranding',
+    'description' => 'Mobile manufacturer or branded name (examples: Samsung, HTC, Verizon, T-Mobile).',
+  ),
+  'mobileDeviceModel' =>
+  array(
+    'type' => 'dimension',
+    'group' => 'Device / Platform',
+    'name' => 'ga:mobileDeviceModel',
+    'description' => 'Mobile device model (example: Nexus S).',
+  ),
+  'mobileInputSelector ' =>
+  array(
+    'type' => 'dimension',
+    'group' => 'Device / Platform',
+    'name' => 'ga:mobileInputSelector',
+    'description' => 'Selector used on the mobile device (examples: touchscreen, joystick, click-wheel, stylus).',
+  ),
+  'mobileDeviceInfo' =>
+  array(
+    'type' => 'dimension',
+    'group' => 'Device / Platform',
+    'name' => 'ga:mobileDeviceInfo',
+    'description' => 'The branding, model, and marketing name used to identify the mobile device.',
+  ),
   'visitors' => 
   array(
     'type' => 'metric',
@@ -1248,6 +1451,62 @@ function google_analytics_views_get_fields() {
     'name' => 'ga:itemsPerPurchase',
     'description' => '<code>ga:itemQuantity / ga:uniquePurchases</code><br/>The average quantity of this item (or group of items) sold per purchase.',
   ),
+  'socialActivities' =>
+  array(
+    'type' => 'metric',
+    'group' => 'Social Activities',
+    'name' => 'ga:socialActivities',
+    'description' => 'The count of activities where a content URL was shared/mentioned on a social data hub partner network.',
+  ),
+  'socialInteractions' =>
+  array(
+    'type' => 'metric',
+    'group' => 'Social Interactions',
+    'name' => 'ga:socialInteractions',
+    'description' => 'The total number of social interactions on your site.',
+  ),
+  'uniqueSocialInteractions' =>
+  array(
+    'type' => 'metric',
+    'group' => 'Social Interactions',
+    'name' => 'ga:uniqueSocialInteractions',
+    'description' => 'The total number of unique social interactions on your site. Based on the the value pair ga:socialInteractionNetwork and ga:socialInteractionAction.',
+  ),
+  'socialInteractionsPerVisit' =>
+  array(
+    'type' => 'metric',
+    'group' => 'Social Interactions',
+    'name' => 'ga:socialInteractionsPerVisit',
+    'description' => '<code>ga:socialInteractions / ga:uniqueSocialInteractions</code><br/>The average number of social interactions per visit to your site.',
+  ),
+  'exceptions' =>
+  array(
+    'type' => 'metric',
+    'group' => 'Exception Tracking',
+    'name' => 'ga:exceptions',
+    'description' => 'The number of exceptions that were sent to Google Analytics.',
+  ),
+  'fatalExceptions' =>
+  array(
+    'type' => 'metric',
+    'group' => 'Exception Tracking',
+    'name' => 'ga:fatalExceptions',
+    'description' => 'The number of exceptions where isFatal is set to true.',
+  ),
+  'userTimingSample' =>
+  array(
+    'type' => 'metric',
+    'group' => 'User Timings',
+    'name' => 'ga:userTimingSample',
+    'description' => 'The number of samples that were sent.',
+  ),
+  'avgUserTimingValue' =>
+  array(
+    'type' => 'metric',
+    'group' => 'User Timings',
+    'name' => 'ga:avgUserTimingValue',
+    'description' => '<code>ga:userTimiingValue / ga:userTimiingSample</code><br/>The average user timing value.',
+  ),
 );
 
 }
